e2e/toolkit/log.go

```diff
@@ -0,0 +1,37 @@
+package toolkit
+
+import (
+	"context"
+	"log"
+	"testing"
+)
+
+// testLoggerKey is a private key to prevent *testing.T from being easily accessed
+type testLoggerKey struct{}
+
+func ContextWithLog(ctx context.Context, t *testing.T) context.Context {
```

Comment:
I understand that `*testing.T` is technically a "logger" since it defined `Logf`, though I might consider changing this to `ContextWithT` since we're adding/popping *testing.T's on/off the context

Reply:
something like `type testingLogger interface { ... } `

Reply:
I know it's ugly. And I had issues with it in the past.
But in this case it's only used for testing, so the worst case a test will fail.

Initially I just tried to pass `logfunc` instead of `*testing.T`. But it's been quite annoying.

I think it's ok to pass tracing information through context that will never be using for any control flow.